src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, hash);
+            size++;
+        } else {
+            if (table[index].value == value) {
+                return;
+            }
+            if (table[index].key == key) {
+                table[index].value = value;
+                return;
+            }
+            Node<K, V> current = table[index];
+            while (current.next != null) {
+                if ((key == null && current.key == null) || (key != null && key.equals(current.key))) {
+                    current.value = value;
+                    return;
+                }
+                current = current.next;
+            }
+            current.next = new Node<>(key, value, hash);
+            size++;
+        }
+        if (size > table.length * LOAD_FACTOR) {
+            resizeTable();
+        }
     }
 
-    @Override
     public V getValue(K key) {
+        int hash = (key == null) ? 0 : key.hashCode();
+        int index = (key == null) ? 0 : getIndex(hash, table.length);
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if ((key == null && current.key == null) || (key != null && key.equals(current.key))) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
-    @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(int hash, int tableLength) {
+        return (hash & Integer.MAX_VALUE) % tableLength;
+    }
+
+    private void resizeTable() {
+        int newCapacity = table.length * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
+        for (Node<K, V> oldNode : table) {
+            while (oldNode != null) {
+                Node<K, V> next = oldNode.next;
+                int newIndex = (oldNode.key == null) ? 0 : getIndex(oldNode.hash, newCapacity);
+                oldNode.next = newTable[newIndex];
+                newTable[newIndex] = oldNode;
+                oldNode = next;
+            }
+        }
+        table = newTable;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private final int hash;
+        private Node<K, V> next;
+
+        public Node(K key, V value, int hash) {
+            this.key = key;
+            this.value = value;
+            this.hash = hash;
+            this.next = null;
+        }
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null || getClass() != obj.getClass()) {
+                return false;
+            }
+            Node<?, ?> node = (Node<?, ?>) obj;
+            return Objects.equals(key, node.key) && Objects.equals(value, node.value);
+        }
+
+        @Override
+        public int hashCode() {
```

Comment:
I think that we are hashing only keys. And do we need to hash it there? If we can hash it in the node, why do we need hash field if we can to hash it every time inside? And If we will hash it inside the node, the hash needs to depend from all the fields maybe, so thats why hash will change after changing 1 field. But in our case, the hash final and depends only from keys if we are talking about hash table.